### ISSUE TITLE
Implement multiple user commands in a single CLI command

### ISSUE THREAD

**Issue by milesgardner** [uchicago-cs/chiventure#1396]
team: cli
members: Miles, Matteo, Kate

task: add the ability for the command line to take multiple commands at once i.e. "go south, go east."  The commands would be separated by a symbol such as a comma or semicolon, and the parser would take multiple commands (more than 2).

This functionality would be very useful, as other teams may want to test their changes, and won't have to type in each command individually.  Instead, they would be able to do their entire test in one command separated by semicolons.

**Comment by mrestuccia1** [comment]
We have added the option to use multiple commands separated by the ';' character. This was implemented using an array of pointers to cmd types. This is very similar to a feature the 'and' command, which allows you to input multiple commands separated by 'and'.

**Comment by mrestuccia1** [comment]
Because these were too similar, we changed them slightly, making it such that the 'and' is conditional, meaning that if one of the commands called fail, the rest of the commands will not be executed. On the other hand, commands separated by ';' will execute unconditionally. There are only two current limitations to this. * Firstly, right now there is only a way to check if commands fail to print to the screen, not if commands just overall do not match an action. This could be a good task to implement in the future such that a command is not executed when another command simply isn't found.
Secondly, we have to maually allocate space for the multiple 'and' commands, so a max of 5 commands at a time will be allowed, else the program will seg fault.

**Comment by mrestuccia1** [comment]
Here is an example of this functionality at work:
When two commands are called with 'and', only one is executed because the first command failed (it contained too many tokens). However, when we run the same command with ';', the second command runs, regardless of the fact that the first fails. This will be a useful feature in the game, as there are many actions you might only want to do in a certain order, and if one fails you might now want to do the following action.

![Screenshot (29)](https://user-images.githubusercontent.com/102557721/170844300-6a5f6f93-f4f2-439f-a54a-79e9aad9c826.png)

**Comment by mrestuccia1** [comment]
I encountered a problem with strtok_r, as it was tokenizing correctly, but was somehow editing the output in a way that caused the following functions that were called on it to not recognize it. After doing research, I found that strtok is destructive, so I used a strtok-like helper function (called strtokstr_r) that had been created by previous CLI teams. This fixed the problem, and the program was thus fully working.
